diesel/src/query_builder/delete_statement.rs

```diff
@@ -11,12 +11,16 @@ impl<T> DeleteStatement<T> {
 
 impl<T> QueryFragment for DeleteStatement<T> where
     T: UpdateTarget,
+    T::WhereClause: QueryFragment,
 {
     fn to_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
         out.push_context(Context::Delete);
         out.push_sql("DELETE FROM ");
         try!(self.0.from_clause(out));
-        try!(self.0.where_clause(out));
+        if let Some(clause) =  self.0.where_clause() {
```

Comment:
Fair enough.  I am not a big fan of Tuple Structs because it seems like normal structs could make the code more readable than the implicit integer names.